neuraxle/steps/sklearn.py

```diff
@@ -78,16 +78,25 @@ def transform(self, data_inputs):
             return self.wrapped_sklearn_predictor.predict(data_inputs)
         return self.wrapped_sklearn_predictor.transform(data_inputs)
 
-    def set_hyperparams(self, flat_hyperparams: HyperparameterSamples) -> BaseStep:
-        BaseStep.set_hyperparams(self, flat_hyperparams)
-        self.wrapped_sklearn_predictor.set_params(**HyperparameterSamples(flat_hyperparams).to_flat_as_dict_primitive())
+    def _set_hyperparams(self, hyperparams: HyperparameterSamples) -> BaseStep:
+        """
+        Set hyperparams for base step, and the wrapped sklearn_predictor.
+
+        :param hyperparams:
+        :return: self
+        """
+        # flatten the step hyperparams, and set the wrapped sklearn predictor params
+        hyperparams = HyperparameterSamples(hyperparams)
+        BaseStep._set_hyperparams(self, hyperparams.to_flat())
+        self.wrapped_sklearn_predictor.set_params(**hyperparams.to_flat_as_dict_primitive())
```

Comment:
Should we force the separator not to change for `sklearn.pipeline.Pipeline`s  ?
```suggestion
        self.wrapped_sklearn_predictor.set_params(**hyperparams.with_separator("__").to_flat_as_dict_primitive())
```